spec/datadog/appsec/reactive/operation_spec.rb

```diff
@@ -0,0 +1,87 @@
+# typed: ignore
+# frozen_string_literal: true
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/reactive/operation'
+
+RSpec.describe Datadog::AppSec::Reactive::Operation do
+  after(:each) do
+    Thread.current[:datadog_security_active_operation] = nil
+  end
+
+  context 'initialize' do
+    it 'set current datadog_security_active_operation thread variable for the duration of the block' do
+      active_operation = Thread.current[:datadog_security_active_operation]
```

Comment:
The thread storage is an implementation detail, so I'm thinking this should be a method, to be defined in the scope above.

```
def active_operation
  Thread.current[:datadog_security_active_operation]
end
```

This would turn the example into something higher level, like:

```
it 'sets the active operation for the duration of the block' do
  expect(active_operation).to be_nil
    described_class.new('test') do |op|
      expect(active_operation).to eq(op)
    end
  expect(active_operation).to be_nil
end
```

And of course something similar for the one below.